libs/blocks/merch/merch.js

```diff
@@ -198,22 +198,47 @@ export async function fetchCheckoutLinkConfigs(base = '') {
   return fetchCheckoutLinkConfigs.promise;
 }
 
-export async function getCheckoutLinkConfig(productFamily) {
+/**
+ * @param {*} productFamily product family code
+ * @param {*} productCode product code, can be undefined
+ * @param {*} paCode product arrangement code, can be undefined
+ * @returns checkout link config for the given product family and code
+ */
+export async function getCheckoutLinkConfig(productFamily, productCode, paCode) {
   let { base } = getConfig();
   if (/\.page$/.test(document.location.origin)) {
     /* c8 ignore next 2 */
     base = base.replace('.live', '.page');
   }
   const checkoutLinkConfigs = await fetchCheckoutLinkConfigs(base);
   const { locale: { region } } = getConfig();
-  const productFamilyConfigs = checkoutLinkConfigs.data?.filter(
-    ({ [NAME_PRODUCT_FAMILY]: mappingProductFamily }) => mappingProductFamily === productFamily,
-  );
-  if (productFamilyConfigs.length === 0) return undefined;
-  const checkoutLinkConfig = productFamilyConfigs.find(
+
+  const {
+    paCodeConfigs,
+    productCodeConfigs,
+    productFamilyConfigs,
+  } = checkoutLinkConfigs.data.reduce((acc, config) => {
```

Comment:
The suggested change causing a strange eslint warning.
Since it was `checkoutLinkConfigs.data` before with no error, I'll keep the line as is. 
Instead, I initialize it as `checkoutLinkConfigs = { data: [] }` in case of error.